### ISSUE TITLE
Portal Link to IMG Data

### ISSUE THREAD

**Issue by ssarrafan** [microbiomedata/nmdc-server#369]
Provide hyperlink to same metagenome record in IMG/M 

Priority - High
Urgency - High

**Comment by jbeezley** [comment]
We likely need this information added to the schema and the workflow pipelines.  Once that is done, it's a quick task to add it to the UI.

**Comment by ssarrafan** [comment]
@pvangay and @emileyfadrosh since this is high priority let me know if this should get re-assigned to Bill and team or if it's on hold till June.

**Comment by jeffbaumes** [comment]
We do have the GOLD id if it came from GOLD and could add a link, but for things that did not come from GOLD we wouldn't know what to do.

Eventually we want an explicit external URL field in the schema, but we can do this other quick fix for now if desired.

**Comment by pvangay** [comment]
Agree with @kfagnan about pushing for an external URL field (or multiple) - I think this urgency is probably more like a medium although @emileyfadrosh may disagree :)

**Comment by subdavis** [comment]
Realized today that I don't know how to construct this URL:

Take this for example: https://data.microbiomedata.org/details/sample/gold:Gb0119269

The GOLD ID is Gb0119269

The JGI link is https://gold.jgi.doe.gov/biosample?id=Gb0115840

What should I link to in IMG?  Is this still an issue for me, or is this a schema issue to get the link in mongo and ingest it first?

CC @jeffbaumes

**Comment by ssarrafan** [comment]
Sounds like this won't be done today. I will move this to August.

**Comment by ssarrafan** [comment]
@subdavis let me know if you need help with this from anyone on the NMDC team

**Comment by emileyfadrosh** [comment]
@subdavis is waiting on me, sorry! I will get the list of links posted here next week, very sorry for the delay!

**Comment by emileyfadrosh** [comment]
quick update: I had added a spreadsheet for this other ticket: https://github.com/microbiomedata/NMDC_Planning/issues/83

but this is just a quick-fix, and ideally the IMG taxon_oid would be pulled from the GOLD API. However, we would likely need to put the full url in the schema since GOLD does not maintain the url, only the IMG ID (eg, 3300042813 and not https://img.jgi.doe.gov/cgi-bin/mer/main.cgi?section=TaxonDetail&page=taxonDetail&taxon_oid=3300042813).


@turbomam @wdduncan @cmungall @dehays @ssarrafan @kfagnan - I am not sure if there is a specific ticket for schema updates for theses links?

**Comment by wdduncan** [comment]
@emileyfadrosh as we discussed on metadata call we have an [alternative identifiers](https://microbiomedata.github.io/nmdc-schema/alternative_identifiers/#slot-alternative-identifiers) slot that can hold the ids even if the ids are not web resolvable. Does that suffice for now?

**Comment by wdduncan** [comment]
If more slots are needed, please be sure to post an issue on [nmdc-schema issue tracker](https://github.com/microbiomedata/nmdc-schema/issues) :)

**Comment by emileyfadrosh** [comment]
Ok, great, thank you! And sorry, just to clarify: the alternative identifiers can hold multiple records (eg, proposal IDs, IMG IDs, ESS-DIVE IDs)? Thanks @wdduncan

**Comment by wdduncan** [comment]
@emileyfadrosh Yes, a single record/entity may have multiple alternative identifiers assigned to it. Does answer your question?

**Comment by subdavis** [comment]
Update:  Because Mongo has `igsn` identifiers and the spreadsheet has `gold` identifiers, I can't match most of the records.   I can match some of them enough to make the schema changes on our side, so I'm not blocked, but I won't be able to populate the portal with all the correct data until I have IDs to join on.

Since changes are now required in mongo, it seems like it might be best to just go ahead and ingest the data form https://github.com/microbiomedata/NMDC_Planning/issues/83 into mongo since loading this from a spreadsheet on ingest isn't ideal?

**Comment by dehays** [comment]
@subdavis I'm going to take care of populating study and biosample records in Mongo with the information from Emiley's spreadhsheet from microbiomedata/NMDC_Planning#83  I'll let you know when this has happened and you can retrieve values during Mongo ingest.  You do not need to parse additional spreadhsheets outside of your Mongo metadata ingest process.

Most of these will appear in the alternative_identifiers list.   I can work with you in how to use the identifiers to link to the other repositories (IMG, INSDC/NCBI, ESS-DIVE, etc.)

**Comment by ssarrafan** [comment]
> @subdavis I'm going to take care of populating study and biosample records in Mongo with the information from Emiley's spreadhsheet from [microbiomedata/NMDC_Planning#83](https://github.com/microbiomedata/NMDC_Planning/issues/83) I'll let you know when this has happened and you can retrieve values during Mongo ingest. You do not need to parse additional spreadhsheets outside of your Mongo metadata ingest process.
> 
> Most of these will appear in the alternative_identifiers list. I can work with you in how to use the identifiers to link to the other repositories (IMG, INSDC/NCBI, ESS-DIVE, etc.). Relates to #370 and #371 as well

@dehays and @subdavis I will move this to the September sprint and assign to @dehays.  David once you're done with your part can. you let Brandon know and re-assign this to him please?  If you prefer to have a separate GH issue let me know.

**Comment by ssarrafan** [comment]
@dehays any update on this?  Is the kitware team still waiting for something from you to proceed?

**Comment by dehays** [comment]
We have the values for this - so blocked on microbiomedata/nmdc-runtime#23 which I'm told will be available this week.

@subdavis The IMG identifiers that I will add to the alternative_identifiers attribute on study - will look like this: img.taxon:3300009759  The CURIE prefix img.taxon is registered - so you can use this ID to link by using one of the identifier resolution services.  For example - you can create your URL like this http://identifers.org/img.taxon:3300009759 to link to the right page at IMG.

**Comment by subdavis** [comment]
Excellent, thanks for the update.  If this is the case, I think the new IDs will just appear after the new data ingest.  If they need to be tweaked I can do that.

**Comment by ssarrafan** [comment]
@dehays this is part of the change sheets correct?  If so, can we close this issue?

**Comment by ssarrafan** [comment]
Checked with David on this and he said to leave this open until @subdavis has been able to implement the links to IMG from study pages. @subdavis can you let me know when it's done so I can close this or close it please?  Thank you

Moving to October sprint for now.

**Comment by ssarrafan** [comment]
This appears to be done.  Let me know if it should be reopened @dehays @subdavis
